openmp/CMakeLists.txt

```diff
@@ -113,7 +113,17 @@ option(OPENMP_ENABLE_LIBOMP_PROFILING "Enable time profiling for libomp." OFF)
 
 # Header install location
 if(${OPENMP_STANDALONE_BUILD})
-  set(LIBOMP_HEADERS_INSTALL_PATH "${CMAKE_INSTALL_INCLUDEDIR}")
+  if(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+    execute_process(
+      OUTPUT_STRIP_TRAILING_WHITESPACE
+      COMMAND ${CMAKE_CXX_COMPILER} --print-resource-dir
+      RESULT_VARIABLE COMMAND_RETURN_CODE
+      OUTPUT_VARIABLE COMPILER_RESOURCE_DIR
+    )
+    set(LIBOMP_HEADERS_INSTALL_PATH "${COMPILER_RESOURCE_DIR}/include")
```

Comment:
The addition of this change seems to have broken the stand alone build we were doing:
```
-- Installing: /nfs/build//lib/clang/19/include/omp.h
CMake Error at runtime/src/cmake_install.cmake:101 (file):
  file INSTALL cannot copy file
  "/home/daltenty/build-openmp/32/runtime/src/omp.h" to
  "/nfs/build/lib/clang/19/include/omp.h":
  Read-only file system.
Call Stack (most recent call first):
  runtime/cmake_install.cmake:42 (include)
  cmake_install.cmake:42 (include)
  
 We're now trying to install to the build compiler's resource directory for some reason, which is read-only and isn't really what we intended here.

Reply:
Yeah that's definitely wrong for a standalone build. For a runtimes build it's so we install it in a place where the compiler can find it so `./bin/clang` actually works. We'll need to fix a lot of this.

Reply:
I think there's some confusion here about the effects of this.

Just for terminology clarity, my understanding is there are roughly three ways we can build runtimes subprojects of LLVM (defined by the top-level CMake source directory and CMake options you use):

1. **Projects+Runtimes build** with Clang (i.e. ):
`cmake -S llvm-project/llvm -DENABLE_PROJECTS=clang -DENABLE_RUNTIMES=openmp`
2. **Runtimes-only build**
`cmake -S llvm-project/runtimes -DENABLE_RUNTIMES=openmp`
3. **Stand-alone build** (from before the monorepo existed, doesn't really work for all subprojects anymore)
`cmake -S llvm-project/openmp -DOPENMP_LLVM_TOOLS_DIR=/path/to/llvm/build/bin`

> Yeah that's definitely wrong for a standalone build. For a runtimes build it's so we install it in a place where the compiler can find it so `./bin/clang` actually works. We'll need to fix a lot of this.

I assume this code was intended for a Projects+Runtimes build with clang. But it's inside a `if(OPENMP_STANDALONE_BUILD)` block which only seems to be true for standalone builds. 

There are some misleading comments in the source about this here https://github.com/llvm/llvm-project/blob/3e921d3c52542b3284c4bd0ef29a70511d38c79c/openmp/CMakeLists.txt#L3 but I checked my builds and I don't see it being set by llvm/runtimes at all and seems to be false during the tests. So I'm not sure this doing what was intended.

Reply:
Posted https://github.com/llvm/llvm-project/pull/91243 to address this